subprojects/packagefiles/speexdsp/libspeexdsp/meson.build

```diff
@@ -0,0 +1,53 @@
+sources = [
+  'buffer.c',
+  'fftwrap.c',
+  'filterbank.c',
+  'jitter.c',
+  'kiss_fft.c',
+  'kiss_fftr.c',
+  'mdf.c',
+  'preprocess.c',
+  'resample.c',
+  'scal.c',
+  'smallft.c',
+]
+
+cc_is_not_msvc = cc.get_id() != 'msvc'
+
+m_dep = cc.find_library('m', required : cc_is_not_msvc)
+
+speexdsp = library('speexdsp',
+  sources,
+  dependencies: m_dep,
+  include_directories : incdir,
+  vs_module_defs: '../win32/libspeexdsp.def',
+  version: lib_version,
+  soversion: cc_is_not_msvc ? lib_soversion : '',
+)
+
+speexdsp_dep = declare_dependency(
+  link_with : speexdsp,
+  include_directories : incdir
+)
+
+pkgconfig = import('pkgconfig')
+pkgconfig.generate(speexdsp,
+  description: 'SpeexDSP is a patent-free, Open Source/Free Software DSP library',
+)
+
+# Unit Tests [TODO]
+#  - Figure out how to run all of them. Currently 'denoise',  'echo',
+#    and 'resample' don't run or sit waiting for stdin.
```

Comment:
This is guarded by the upstream autotools as `if BUILD_EXAMPLES` so those may not be intended to work as unittests.

Reply:
Thanks for catching that, @eli-schwartz!
The filename start with `test<feature>.c`, which threw me off. 
I've removed them.

Reply:
IMO you should still build them, they are a useful verification that the programs *build* and especially link correctly against the library.

They just aren't useful as a unittest.

Reply:
Decided to make the wrap more feature-complete with `meson_options`, and added `with-example-programs` to the options.  The executables are also placed inside an `./examples/` directory, to be extra-clear as to what they are.